src/compiler/checker.ts

```diff
@@ -11934,6 +11934,10 @@ namespace ts {
             });
         }
 
+        /**
+         * Caution: Do you _really_ mean to use this fucntion? It creates fresh property symbols in an uncached manner - if you, the caller,
+         * do not cache the result of this function, you may have very poor performance over large unions.

```

Comment:
is there a cached version, or at least a common way to cache its result? If so it would be good to mention that too.

Reply:
Not quite? If you have a union, the property cache on the union serves a similar purpose. I added this comment because I'd used it in the first iteration, and it caused horrendous perf regressions because of the lack of caching it caused.